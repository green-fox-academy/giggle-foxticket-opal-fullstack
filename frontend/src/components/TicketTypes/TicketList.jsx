import React from 'react';
import './TicketList.styles.sass';
import AdminTicket from './AdminTicket';

const ticketTypes = [
  {
    id: 1,
    title: 'Ticket-Types A',
    description: 'Lorem Ipsum has been the industry s standard dummy',
    iconName: 'FaBeer',
  },
  {
    id: 2,
    title: 'Ticket-Types B',
    description: 'Since the 1500s,when an unknown printer took a galley',
    iconName: 'FaRegSmileWink',
  },
  {
    id: 3,
    title: 'Ticket-Types C',
    description: 'Remaining  essentially unchanged. It was popularised',
    iconName: 'FaRegGem',
  },
  {
    id: 4,
    title: 'Ticket-Types D',
    description: 'Contrary to popular Lorem Ipsum is not  random text',
    iconName: 'FaUmbrella',
  },
];

const TicketList = () => {
  return (
    <div className="ticket-list-container">
      <div className="ticket-list">
        <h1 className="main-title">Ticket Types</h1>
        {ticketTypes.map(i => (
          <AdminTicket key={i.id} {...i} />
        ))}
      </div>
    </div>
  );
};

export default TicketList;
