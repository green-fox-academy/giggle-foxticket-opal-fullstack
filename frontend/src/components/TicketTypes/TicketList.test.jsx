import React from 'react';
import TicketList from './TicketList';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<TicketList />).toJSON();
  expect(tree).toMatchSnapshot();
});
