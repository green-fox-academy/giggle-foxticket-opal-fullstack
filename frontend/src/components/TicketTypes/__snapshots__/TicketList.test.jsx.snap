// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ticket-list"
>
  <div
    className="ticket-list-container"
  >
    <h1
      className="ticket-types-header"
    >
      Ticket Types
    </h1>
    <div
      className="ticket"
    >
      <img
        alt="cat"
        className="ticket-image"
        src="ticket-icon.png"
      />
      <div
        className="ticket-description"
      >
        <h3
          className="ticket-header"
        >
          Three-line item
        </h3>
        The above example no longer works, I think due to changes in newer versions of Sass. It cannot handle the multi-line expression (issue here). Each grid row must be defined all in the same line: The above
      </div>
      <button
        className="btn btn--primary--solid btn--medium"
      >
        BUY
      </button>
    </div>
    <div
      className="ticket"
    >
      <img
        alt="cat"
        className="ticket-image"
        src="ticket-icon.png"
      />
      <div
        className="ticket-description"
      >
        <h3
          className="ticket-header"
        >
          Three-line item
        </h3>
        So CSS grid-template-areas ASCII art works fine with .scss syntax, just not .sass after toying with it. Of course .scss is identical to CSS. Was hoping .sass would work to keep it cleaner.So CSS
      </div>
      <button
        className="btn btn--primary--solid btn--medium"
      >
        BUY
      </button>
    </div>
    <div
      className="ticket"
    >
      <img
        alt="cat"
        className="ticket-image"
        src="ticket-icon.png"
      />
      <div
        className="ticket-description"
      >
        <h3
          className="ticket-header"
        >
          Three-line item
        </h3>
        If anyone knows any discussion going on about .sass working here (or secret hacks to make .sass work with the ASCII portion of grid-template-areas), would love to check out... Thanks in a
      </div>
      <button
        className="btn btn--primary--solid btn--medium"
      >
        BUY
      </button>
    </div>
  </div>
</div>
`;
